Sources/Hummingbird/Server/URI+decodeQuery.swift

```diff
@@ -0,0 +1,45 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Hummingbird server framework project
+//
+// Copyright (c) 2024 the Hummingbird authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See hummingbird/CONTRIBUTORS.txt for the list of Hummingbird authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import HummingbirdCore
+
+extension URI {
+    /// Decode request query using ``URLEncodedFormDecoder``.
+    /// - Parameters
+    ///   - type: Type you want to decode to
+    ///   - context: Request context
+    public func decodeQuery<Type: Decodable>(as type: Type.Type, context: some RequestContext) throws -> Type {
```

Comment:
Idea: We could add add type inferrence here:
```suggestion
    public func decodeQuery<Type: Decodable>(as type: Type.Type = Type.self, context: some RequestContext) throws -> Type {
```